packages/contracts-bedrock/src/libraries/SemverComp.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Libraries
+import { LibString } from "solady/src/utils/LibString.sol";
+import { JSONParserLib } from "solady/src/utils/JSONParserLib.sol";
+
+/// @notice Library for comparing semver strings. Ignores prereleases and build metadata.
```

Comment:
Let's clearly document the threat model for this library - it's deliberately not perfect against malformed semvers, it assumes the input is relatively well formed.